spec/controllers/users_controller_spec.rb

```diff
@@ -40,5 +40,38 @@
       end
       
     end
+
+    context "sending emails" do
+      
+      after { ActionMailer::Base.deliveries.clear }
+
+      it "sends out email to the user with valid inputs" do
```

Comment:
`with_valid_inputs` repeat at all your below specs. I recommend pushing it under upper context where you handle `valid_inputs` with context `sending emails`